test/jest/store.flags.spec.ts

```diff
@@ -0,0 +1,86 @@
+import Store, { IStore } from '../../src/store';
+import { IMParticleWebSDKInstance } from '../../src/mp-instance';
+import { SDKInitConfig } from '../../src/sdkRuntimeModels';
+
+describe('Store privacy flags', () => {
+    let store: IStore & Record<string, any>;
+    let mockMPInstance: IMParticleWebSDKInstance & Record<string, any>;
+
+    beforeEach(() => {
+        mockMPInstance = {
+            _Helpers: {
+                createMainStorageName: () => 'mprtcl-v4',
+                createProductStorageName: () => 'mprtcl-prodv4',
+                Validators: { isFunction: () => true },
+                extend: Object.assign,
+            },
+            _NativeSdkHelpers: {
+                isWebviewEnabled: () => false,
+            },
+            _Persistence: {
+                update: () => {},
+                savePersistence: () => {},
+                getPersistence: () => ({ gs: {} }),
+            },
+            Logger: {
+                verbose: () => {},
+                warning: () => {},
+                error: () => {},
+            },
+            Identity: {
+                getCurrentUser: () => ({ getMPID: () => 'mpid' }),
+            },
+        } as any;
+
+        store = {} as any;
+        // Construct Store with our mock 'this'
+        (Store as any).call(store, {} as SDKInitConfig, mockMPInstance, 'apikey');
+    });
+
+    // Default case
+    it('defaults to false when not provided', () => {
```

Comment:
i'd be more specific


```suggestion
    it('defaults noFunctional and noTargeting to false when not provided', () => {
```